File: DuDoong-Api/src/main/java/band/gosrock/api/comment/model/response/CreateCommentResponse.java
package band.gosrock.api.comment.model.response;


import band.gosrock.common.annotation.DateFormat;
import band.gosrock.domain.common.vo.UserInfoVo;
import band.gosrock.domain.domains.comment.domain.Comment;
import band.gosrock.domain.domains.user.domain.User;
import java.time.LocalDateTime;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class CreateCommentResponse {

    private final Long id;

    private final String nickName;

    private final String content;

    @DateFormat private final LocalDateTime createdAt;

    private final UserInfoVo userInfoVo;

    public static CreateCommentResponse of(Comment comment, User user) {
        return CreateCommentResponse.builder()
                .id(comment.getId())
                .nickName(comment.getNickName())
                .content(comment.getContent())
                .createdAt(comment.getCreatedAt())
                .userInfoVo(user.toUserInfoVo())
                .build();
    }
}

File: DuDoong-Api/src/main/kotlin/band/gosrock/api/comment/model/response/CreateCommentResponse.kt
package band.gosrock.api.comment.model.response

import band.gosrock.common.annotation.DateFormat
import band.gosrock.domain.common.vo.UserInfoVo
import band.gosrock.domain.domains.comment.domain.Comment
import band.gosrock.domain.domains.user.domain.User
import java.time.LocalDateTime

data class CreateCommentResponse(
    val id: Long?,
    val nickName: String?,
    val content: String?,
    @DateFormat val createdAt: LocalDateTime?,
    val userInfoVo: UserInfoVo?,
) {
    companion object {
        @JvmStatic
        fun of(comment: Comment, user: User): CreateCommentResponse =
            CreateCommentResponse(
                id = comment.id,
                nickName = comment.nickName,
                content = comment.content,
                createdAt = comment.createdAt,
                userInfoVo = user.toUserInfoVo(),
            )
    }
}
